File: src/VB2019-parts-Import/登録クラスVisualBasic2019/source/clsImportSK.vb
Option Strict Off
Option Explicit On
Option Compare Binary
Friend Class clsImportSK
	' @(h) clsReadSK.cls
	' @(s)
	' JVData "SK" データベース登録クラス
	'
	
	Private mBuf As JV_SK_SANKU '' 産駒マスタ構造体
	Private mRS As ADODB.Recordset
	
	
	' @(f)
	'
	' 機能      : 初期処理
	'
	' 引き数    :
	'
	' 返り値    :
	'
	' 機能説明  :
	'

    Private Sub Class_Initialize_Renamed()
        On Error GoTo ErrorHandler

        Dim strSql As String ''SQL文

        'レコードセットオープン
        strSql = "SELECT * FROM SANKU"
        mRS = New ADODB.Recordset()
        mRS.Open(strSql, gCon, ADODB.CursorTypeEnum.adOpenForwardOnly, ADODB.LockTypeEnum.adLockPessimistic)

ExitHandler:
        Exit Sub
ErrorHandler:
        System.Diagnostics.Debug.WriteLine(Err.Description)
        Resume ExitHandler
    End Sub
    Public Sub New()
        MyBase.New()
        Class_Initialize_Renamed()
    End Sub
    ' @(f)
    '
    ' 機能      : Closeのコーディング
    '
    ' 機能説明  : ガーベッジコレクションにCloseを呼ばせるとどこで呼ばれるか分からない為、
    '           　明示的に呼び出す必要がある。
    '
    Public Sub Close()
        'レコードセットクローズ
        mRS.Close()

        mRS = Nothing

    End Sub

    ' @(f)
    '
    ' 機能      : 終了処理
    '
    ' 引き数    :
    '
    ' 返り値    :
    '
    ' 機能説明  : レコードセットクローズ
    '

    Private Sub Class_Terminate_Renamed()
        On Error GoTo ErrorHandler

ExitHandler:
        Exit Sub
ErrorHandler:
        System.Diagnostics.Debug.WriteLine(Err.Description)
        Resume ExitHandler
    End Sub
    Protected Overrides Sub Finalize()
        Class_Terminate_Renamed()
        MyBase.Finalize()
    End Sub


    ' @(f)
    '
    ' 機能      : Addプロシージャを呼ぶ
    '
    ' 引き数    : lBuf - JVData 識別子"SK" の１行
    '
    ' 返り値    : True - 成功, False - 失敗
    '
    ' 機能説明  : clsIReadインターフェイスAddプロシージャの実装
    '
    Public Function Add(ByRef strBuf As String, ByVal lngBufSize As Integer) As Boolean
        On Error GoTo ErrorHandler

        Dim strMakeDate As String '' 登録するデータの作成年月日

        '構造体にデータセット
        mBuf.SetData(strBuf)


        With mBuf.head.MakeDate

            strMakeDate = .Year & .Month & .Day
        End With

        'INSERT処理
        If Not InsertDB() Then
            'UPDATE処理（INSERTが失敗した場合）
            If Not UpdateDB(strMakeDate) Then System.Diagnostics.Debug.WriteLine("更新に失敗しました。" & Left(strBuf, 2))
        End If

        Add = True

ExitHandler:
        Exit Function
ErrorHandler:
        Add = False
        System.Diagnostics.Debug.WriteLine(Err.Description)
        Resume ExitHandler
    End Function

    ' @(f)
    '
    ' 機能      : データベースに追加する
    '
    ' 引き数    :
    '
    ' 返り値    : True - 成功, False - 失敗
    '
    ' 機能説明  :
    '
    Public Function InsertDB() As Boolean
        On Error GoTo ErrorHandler
        Dim i As Short '' ループカウンタ
        Dim j As Short '' ループカウンタ
        Dim k As Short '' ループカウンタ

        gCon.BeginTrans()
        System.Diagnostics.Debug.WriteLine("BeginTrans")

        mRS.AddNew()

        With mBuf
            With .head
                mRS.Fields("RecordSpec").Value = .RecordSpec '' レコード種別
                mRS.Fields("DataKubun").Value = .DataKubun '' データ区分
                With .MakeDate
                    mRS.Fields("MakeDate").Value = .Year & .Month & .Day '' 年月日
                End With ' MakeDate
            End With ' head
            mRS.Fields("KettoNum").Value = .KettoNum '' 血統登録番号
            With .BirthDate
                mRS.Fields("BirthDate").Value = .Year & .Month & .Day '' 年月日
            End With ' BirthDate
            mRS.Fields("SexCD").Value = .SexCD '' 性別コード
            mRS.Fields("HinsyuCD").Value = .HinsyuCD '' 品種コード
            mRS.Fields("KeiroCD").Value = .KeiroCD '' 毛色コード
            mRS.Fields("SankuMochiKubun").Value = .SankuMochiKubun '' 産駒持込区分
            mRS.Fields("ImportYear").Value = .ImportYear '' 輸入年
            mRS.Fields("BreederCode").Value = .BreederCode '' 生産者コード
            mRS.Fields("SanchiName").Value = .SanchiName '' 産地名
            mRS.Fields("FNum").Value = .HansyokuNum(0)
            mRS.Fields("MNum").Value = .HansyokuNum(1)
            mRS.Fields("FFNum").Value = .HansyokuNum(2)
            mRS.Fields("FMNum").Value = .HansyokuNum(3)
            mRS.Fields("MFNum").Value = .HansyokuNum(4)
            mRS.Fields("MMNum").Value = .HansyokuNum(5)
            mRS.Fields("FFFNum").Value = .HansyokuNum(6)
            mRS.Fields("FFMNum").Value = .HansyokuNum(7)
            mRS.Fields("FMFNum").Value = .HansyokuNum(8)
            mRS.Fields("FMMNum").Value = .HansyokuNum(9)
            mRS.Fields("MFFNum").Value = .HansyokuNum(10)
            mRS.Fields("MFMNum").Value = .HansyokuNum(11)
            mRS.Fields("MMFNum").Value = .HansyokuNum(12)
            mRS.Fields("MMMNum").Value = .HansyokuNum(13)
        End With

        With mBuf
            System.Diagnostics.Debug.WriteLine("Insert SANKU : " & .KettoNum)
        End With ' id

        mRS.Update()

        gCon.CommitTrans()
        System.Diagnostics.Debug.WriteLine("CommitTrans")

        InsertDB = True

ExitHandler:
        Exit Function
ErrorHandler:
        InsertDB = False
        mRS.CancelUpdate()
        gCon.RollbackTrans()
        System.Diagnostics.Debug.WriteLine(Err.Description)
        System.Diagnostics.Debug.WriteLine("RollbackTrans")
        Resume ExitHandler
    End Function


    ' @(f)
    '
    ' 機能      : データベースを更新する
    '
    ' 引き数    :
    '
    ' 返り値    : True - 成功, False - 失敗
    '
    ' 機能説明  :
    '
    Public Function UpdateDB(ByRef strMakeDate As String) As Boolean
        On Error GoTo ErrorHandler
        Dim i As Short '' ループカウンタ
        Dim j As Short '' ループカウンタ
        Dim k As Short '' ループカウンタ
        Dim strSql As String '' SQL文

        gCon.BeginTrans()
        System.Diagnostics.Debug.WriteLine("BeginTrans")

        strSql = "UPDATE SANKU SET "
        With mBuf
            With .head
                strSql = strSql & SS & "RecordSpec" & SE & "='" & Replace(.RecordSpec, "'", "''") & "'," '' レコード種別
                strSql = strSql & SS & "DataKubun" & SE & "='" & Replace(.DataKubun, "'", "''") & "'," '' データ区分
                strSql = strSql & SS & "MakeDate" & SE & "='" & Replace(strMakeDate, "'", "''") & "'," '' 年月日
            End With ' head
            strSql = strSql & SS & "KettoNum" & SE & "='" & Replace(.KettoNum, "'", "''") & "'," '' 血統登録番号
            With .BirthDate
                strSql = strSql & SS & "BirthDate" & SE & "='" & Replace(.Year & .Month & .Day, "'", "''") & "'," '' 年月日
            End With ' BirthDate
            strSql = strSql & SS & "SexCD" & SE & "='" & Replace(.SexCD, "'", "''") & "'," '' 性別コード
            strSql = strSql & SS & "HinsyuCD" & SE & "='" & Replace(.HinsyuCD, "'", "''") & "'," '' 品種コード
            strSql = strSql & SS & "KeiroCD" & SE & "='" & Replace(.KeiroCD, "'", "''") & "'," '' 毛色コード
            strSql = strSql & SS & "SankuMochiKubun" & SE & "='" & Replace(.SankuMochiKubun, "'", "''") & "'," '' 産駒持込区分
            strSql = strSql & SS & "ImportYear" & SE & "='" & Replace(.ImportYear, "'", "''") & "'," '' 輸入年
            strSql = strSql & SS & "BreederCode" & SE & "='" & Replace(.BreederCode, "'", "''") & "'," '' 生産者コード
            strSql = strSql & SS & "SanchiName" & SE & "='" & Replace(.SanchiName, "'", "''") & "'," '' 産地名
            strSql = strSql & SS & "FNum" & SE & "='" & Replace(.HansyokuNum(0), "'", "''") & "'," '' 父繁殖登録番号
            strSql = strSql & SS & "MNum" & SE & "='" & Replace(.HansyokuNum(1), "'", "''") & "'," '' 母繁殖登録番号
            strSql = strSql & SS & "FFNum" & SE & "='" & Replace(.HansyokuNum(2), "'", "''") & "'," '' 父父繁殖登録番号
            strSql = strSql & SS & "FMNum" & SE & "='" & Replace(.HansyokuNum(3), "'", "''") & "'," '' 父母繁殖登録番号
            strSql = strSql & SS & "MFNum" & SE & "='" & Replace(.HansyokuNum(4), "'", "''") & "'," '' 母父繁殖登録番号
            strSql = strSql & SS & "MMNum" & SE & "='" & Replace(.HansyokuNum(5), "'", "''") & "'," '' 母母繁殖登録番号
            strSql = strSql & SS & "FFFNum" & SE & "='" & Replace(.HansyokuNum(6), "'", "''") & "'," '' 父父父繁殖登録番号
            strSql = strSql & SS & "FFMNum" & SE & "='" & Replace(.HansyokuNum(7), "'", "''") & "'," '' 父父母繁殖登録番号
            strSql = strSql & SS & "FMFNum" & SE & "='" & Replace(.HansyokuNum(8), "'", "''") & "'," '' 父母父繁殖登録番号
            strSql = strSql & SS & "FMMNum" & SE & "='" & Replace(.HansyokuNum(9), "'", "''") & "'," '' 父母母繁殖登録番号
            strSql = strSql & SS & "MFFNum" & SE & "='" & Replace(.HansyokuNum(10), "'", "''") & "'," '' 母父父繁殖登録番号
            strSql = strSql & SS & "MFMNum" & SE & "='" & Replace(.HansyokuNum(11), "'", "''") & "'," '' 母父母繁殖登録番号
            strSql = strSql & SS & "MMFNum" & SE & "='" & Replace(.HansyokuNum(12), "'", "''") & "'," '' 母母父繁殖登録番号
            strSql = strSql & SS & "MMMNum" & SE & "='" & Replace(.HansyokuNum(13), "'", "''") & "'," '' 母母母繁殖登録番号

            strSql = Left(strSql, Len(strSql) - 1) ' Delete Last Comma

            strSql = strSql & " WHERE " & SS & "KettoNum" & SE & "='" & Replace(.KettoNum, "'", "''") & "'"
            strSql = strSql & " AND " & SS & "MakeDate" & SE & " <= '" & Replace(strMakeDate, "'", "''") & "'"
        End With
        With mBuf
            System.Diagnostics.Debug.WriteLine("UPDATE SANKU : " & .KettoNum)
        End With ' id

        gCon.Execute(strSql)

        gCon.CommitTrans()
        System.Diagnostics.Debug.WriteLine("CommitTrans")

        UpdateDB = True

ExitHandler:
        Exit Function
ErrorHandler:
        System.Diagnostics.Debug.WriteLine(Err.Description)
        System.Diagnostics.Debug.WriteLine("RollbackTrans")
        gCon.RollbackTrans()
        UpdateDB = False
        Resume ExitHandler
    End Function
End Class

File: src/VC2019-parts-Import/登録クラスVC2019/clsImportSK.cpp
/*=======================================================================
  JRA-VAN Data Lab.プログラミングパーツ「データ登録パーツ　SEレコード登録クラス」

	   作成: JRA-VAN ソフトウェア工房  2003年6月26日

========================================================================
   (C) Copyright JRA SYSTEM SERVICE CO.,LTD. 2007 All rights reserved
========================================================================*/
#include "StdAfx.h"
#include "clsImportSK.h"

clsImportSK::clsImportSK(void)
{
}

clsImportSK::~clsImportSK(void)
{
}
int clsImportSK::Add(CString strBuff, long lngBuffSize)
{
	CString strMakeDate;
    memcpy(&mBuf,strBuff.GetBuffer(lngBuffSize),lngBuffSize);
    strMakeDate.SetString(mBuf.head.MakeDate.Year,4);
	strMakeDate.Append(mBuf.head.MakeDate.Month,2);
	strMakeDate.Append(mBuf.head.MakeDate.Day,2);

//      INSERT処理
	if(InsertDB() != 0 ){
//			UPDATE処理（INSERTが失敗した場合）
		if(UpdateDB(strMakeDate)!=0){
//				System.Diagnostics.Debug.WriteLine("更新に失敗しました。" & Left(strBuf, 2))
		}
	}
	return 0;

	return 0;
}

int clsImportSK::InsertDB(void)
{
	CString strsql;
	strsql="SELECT * FROM SANKU WHERE [KettoNum] = '";
	strsql.Append(mBuf.KettoNum,10);
	strsql.Append("'");
	_bstr_t bstrQuery(strsql);
    // Commandオブジェクト
	_CommandPtr pCommand;
	_RecordsetPtr pRecordSet;

	pCommand.CreateInstance(__uuidof(Command));
	pRecordSet.CreateInstance(__uuidof(Recordset));

	// SQLの設定
	pCommand->ActiveConnection = pCn;
	pCommand->CommandText = bstrQuery;
	pRecordSet->PutRefSource(pCommand);


	_variant_t vNull;  // VARIANT型のNULLとして使う
	vNull.vt = VT_ERROR;
	vNull.scode = DISP_E_PARAMNOTFOUND;

	// レコードセットの取得
	pRecordSet->Open(vNull, vNull, adOpenKeyset, adLockOptimistic, adCmdText);

	if (!pRecordSet->GetadoEOF()){
		pRecordSet->Close();
		pRecordSet = NULL;
		return -1;
	}

	USES_CONVERSION;

	COleSafeArray vaFieldlist;
	vaFieldlist.CreateOneDim(VT_VARIANT,26);

	COleSafeArray vaValuelist;
	vaValuelist.CreateOneDim(VT_VARIANT,26);
	long lArrayIndex[1];
	
try{
	
	lArrayIndex[0]=0;
	vaFieldlist.PutElement(lArrayIndex, &(_variant_t("RecordSpec")));	
	strsql.SetString(mBuf.head.RecordSpec,2);
	vaValuelist.PutElement(lArrayIndex, &(_variant_t(T2OLE(strsql))));

	lArrayIndex[0]=1;
	vaFieldlist.PutElement(lArrayIndex, &(_variant_t("DataKubun")));	
	strsql.SetString(mBuf.head.DataKubun,1);
	vaValuelist.PutElement(lArrayIndex, &(_variant_t(T2OLE(strsql))));

	lArrayIndex[0]=2;
	vaFieldlist.PutElement(lArrayIndex, &(_variant_t("MakeDate")));	
	strsql.SetString(mBuf.head.MakeDate.Year,8);
	vaValuelist.PutElement(lArrayIndex, &(_variant_t(T2OLE(strsql))));

	lArrayIndex[0]=3;
	vaFieldlist.PutElement(lArrayIndex, &(_variant_t("KettoNum")));	
	strsql.SetString(mBuf.KettoNum,10);
	vaValuelist.PutElement(lArrayIndex, &(_variant_t(T2OLE(strsql))));

	lArrayIndex[0]=4;
	vaFieldlist.PutElement(lArrayIndex, &(_variant_t("BirthDate")));	
	strsql.SetString(mBuf.BirthDate.Year,8);
	vaValuelist.PutElement(lArrayIndex, &(_variant_t(T2OLE(strsql))));

	lArrayIndex[0]=5;
	vaFieldlist.PutElement(lArrayIndex, &(_variant_t("SexCD")));	
	strsql.SetString(mBuf.SexCD,1);
	vaValuelist.PutElement(lArrayIndex, &(_variant_t(T2OLE(strsql))));

	lArrayIndex[0]=6;
	vaFieldlist.PutElement(lArrayIndex, &(_variant_t("HinsyuCD")));	
	strsql.SetString(mBuf.HinsyuCD,1);
	vaValuelist.PutElement(lArrayIndex, &(_variant_t(T2OLE(strsql))));

	lArrayIndex[0]=7;
	vaFieldlist.PutElement(lArrayIndex, &(_variant_t("KeiroCD")));	
	strsql.SetString(mBuf.KeiroCD,2);
	vaValuelist.PutElement(lArrayIndex, &(_variant_t(T2OLE(strsql))));

	lArrayIndex[0]=8;
	vaFieldlist.PutElement(lArrayIndex, &(_variant_t("SankuMochiKubun")));	
	strsql.SetString(mBuf.SankuMochiKubun,1);
	vaValuelist.PutElement(lArrayIndex, &(_variant_t(T2OLE(strsql))));

	lArrayIndex[0]=9;
	vaFieldlist.PutElement(lArrayIndex, &(_variant_t("ImportYear")));	
	strsql.SetString(mBuf.ImportYear,4);
	vaValuelist.PutElement(lArrayIndex, &(_variant_t(T2OLE(strsql))));

	lArrayIndex[0]=10;
	vaFieldlist.PutElement(lArrayIndex, &(_variant_t("BreederCode")));	
	strsql.SetString(mBuf.BreederCode,8);
	vaValuelist.PutElement(lArrayIndex, &(_variant_t(T2OLE(strsql))));

	lArrayIndex[0]=11;
	vaFieldlist.PutElement(lArrayIndex, &(_variant_t("SanchiName")));	
	strsql.SetString(mBuf.SanchiName,20);
	vaValuelist.PutElement(lArrayIndex, &(_variant_t(T2OLE(strsql))));

	lArrayIndex[0]=12;
	vaFieldlist.PutElement(lArrayIndex, &(_variant_t("FNum")));	
	strsql.SetString(mBuf.HansyokuNum[0],10);
	vaValuelist.PutElement(lArrayIndex, &(_variant_t(T2OLE(strsql))));

	lArrayIndex[0]=13;
	vaFieldlist.PutElement(lArrayIndex, &(_variant_t("MNum")));	
	strsql.SetString(mBuf.HansyokuNum[1],10);
	vaValuelist.PutElement(lArrayIndex, &(_variant_t(T2OLE(strsql))));

	lArrayIndex[0]=14;
	vaFieldlist.PutElement(lArrayIndex, &(_variant_t("FFNum")));	
	strsql.SetString(mBuf.HansyokuNum[2],10);
	vaValuelist.PutElement(lArrayIndex, &(_variant_t(T2OLE(strsql))));

	lArrayIndex[0]=15;
	vaFieldlist.PutElement(lArrayIndex, &(_variant_t("FMNum")));	
	strsql.SetString(mBuf.HansyokuNum[3],10);
	vaValuelist.PutElement(lArrayIndex, &(_variant_t(T2OLE(strsql))));

	lArrayIndex[0]=16;
	vaFieldlist.PutElement(lArrayIndex, &(_variant_t("MFNum")));
	strsql.SetString(mBuf.HansyokuNum[4],10);
	vaValuelist.PutElement(lArrayIndex, &(_variant_t(T2OLE(strsql))));

	lArrayIndex[0]=17;
	vaFieldlist.PutElement(lArrayIndex, &(_variant_t("MMNum")));
	strsql.SetString(mBuf.HansyokuNum[5],10);
	vaValuelist.PutElement(lArrayIndex, &(_variant_t(T2OLE(strsql))));

	lArrayIndex[0]=18;
	vaFieldlist.PutElement(lArrayIndex, &(_variant_t("FFFNum")));
	strsql.SetString(mBuf.HansyokuNum[6],10);
	vaValuelist.PutElement(lArrayIndex, &(_variant_t(T2OLE(strsql))));

	lArrayIndex[0]=19;
	vaFieldlist.PutElement(lArrayIndex, &(_variant_t("FFMNum")));
	strsql.SetString(mBuf.HansyokuNum[7],10);
	vaValuelist.PutElement(lArrayIndex, &(_variant_t(T2OLE(strsql))));

	lArrayIndex[0]=20;
	vaFieldlist.PutElement(lArrayIndex, &(_variant_t("FMFNum")));
	strsql.SetString(mBuf.HansyokuNum[8],10);
	vaValuelist.PutElement(lArrayIndex, &(_variant_t(T2OLE(strsql))));

	lArrayIndex[0]=21;
	vaFieldlist.PutElement(lArrayIndex, &(_variant_t("FMMNum")));
	strsql.SetString(mBuf.HansyokuNum[9],10);
	vaValuelist.PutElement(lArrayIndex, &(_variant_t(T2OLE(strsql))));

	lArrayIndex[0]=22;
	vaFieldlist.PutElement(lArrayIndex, &(_variant_t("MFFNum")));
	strsql.SetString(mBuf.HansyokuNum[10],10);
	vaValuelist.PutElement(lArrayIndex, &(_variant_t(T2OLE(strsql))));

	lArrayIndex[0]=23;
	vaFieldlist.PutElement(lArrayIndex, &(_variant_t("MFMNum")));	
	strsql.SetString(mBuf.HansyokuNum[11],10);
	vaValuelist.PutElement(lArrayIndex, &(_variant_t(T2OLE(strsql))));

	lArrayIndex[0]=24;
	vaFieldlist.PutElement(lArrayIndex, &(_variant_t("MMFNum")));	
	strsql.SetString(mBuf.HansyokuNum[12],10);
	vaValuelist.PutElement(lArrayIndex, &(_variant_t(T2OLE(strsql))));

	lArrayIndex[0]=25;
	vaFieldlist.PutElement(lArrayIndex, &(_variant_t("MMMNum")));	
	strsql.SetString(mBuf.HansyokuNum[13],10);
	vaValuelist.PutElement(lArrayIndex, &(_variant_t(T2OLE(strsql))));
	pRecordSet->AddNew(vaFieldlist, vaValuelist);
	pRecordSet->Close();	

}
catch(_com_error &e){
	MessageBox(NULL,e.Description(),NULL,NULL);
	pRecordSet->Close();
	return -1;
}

	return 0;

}

int clsImportSK::UpdateDB(CString strMakeDate)
{
	CString strsql;
	strsql="SELECT * FROM SANKU WHERE [KettoNum] = '";
	strsql.Append(mBuf.KettoNum,10);
	strsql.Append("' AND [MakeDate] <= '");
	strsql.Append(strMakeDate);
	strsql.Append("'");

	_bstr_t bstrQuery(strsql);
    // Commandオブジェクト
	_CommandPtr pCommand;
	_RecordsetPtr pRecordSet;

	pCommand.CreateInstance(__uuidof(Command));
	pRecordSet.CreateInstance(__uuidof(Recordset));

	// SQLの設定
	pCommand->ActiveConnection = pCn;
	pCommand->CommandText = bstrQuery;
	pRecordSet->PutRefSource(pCommand);


	_variant_t vNull;  // VARIANT型のNULLとして使う
	vNull.vt = VT_ERROR;
	vNull.scode = DISP_E_PARAMNOTFOUND;

	// レコードセットの取得
	pRecordSet->Open(vNull, vNull, adOpenKeyset, adLockOptimistic, adCmdText);

	if (!pRecordSet->GetadoEOF()){
		pRecordSet->Close();
		pRecordSet = NULL;
		return -1;
	}

	USES_CONVERSION;

	COleSafeArray vaFieldlist;
	vaFieldlist.CreateOneDim(VT_VARIANT,26);

	COleSafeArray vaValuelist;
	vaValuelist.CreateOneDim(VT_VARIANT,26);
	long lArrayIndex[1];
	
try{
	
	lArrayIndex[0]=0;
	vaFieldlist.PutElement(lArrayIndex, &(_variant_t("RecordSpec")));	
	strsql.SetString(mBuf.head.RecordSpec,2);
	vaValuelist.PutElement(lArrayIndex, &(_variant_t(T2OLE(strsql))));

	lArrayIndex[0]=1;
	vaFieldlist.PutElement(lArrayIndex, &(_variant_t("DataKubun")));	
	strsql.SetString(mBuf.head.DataKubun,1);
	vaValuelist.PutElement(lArrayIndex, &(_variant_t(T2OLE(strsql))));

	lArrayIndex[0]=2;
	vaFieldlist.PutElement(lArrayIndex, &(_variant_t("MakeDate")));	
	strsql.SetString(mBuf.head.MakeDate.Year,8);
	vaValuelist.PutElement(lArrayIndex, &(_variant_t(T2OLE(strsql))));

	lArrayIndex[0]=3;
	vaFieldlist.PutElement(lArrayIndex, &(_variant_t("KettoNum")));	
	strsql.SetString(mBuf.KettoNum,10);
	vaValuelist.PutElement(lArrayIndex, &(_variant_t(T2OLE(strsql))));

	lArrayIndex[0]=4;
	vaFieldlist.PutElement(lArrayIndex, &(_variant_t("BirthDate")));	
	strsql.SetString(mBuf.BirthDate.Year,8);
	vaValuelist.PutElement(lArrayIndex, &(_variant_t(T2OLE(strsql))));

	lArrayIndex[0]=5;
	vaFieldlist.PutElement(lArrayIndex, &(_variant_t("SexCD")));	
	strsql.SetString(mBuf.SexCD,1);
	vaValuelist.PutElement(lArrayIndex, &(_variant_t(T2OLE(strsql))));

	lArrayIndex[0]=6;
	vaFieldlist.PutElement(lArrayIndex, &(_variant_t("HinsyuCD")));	
	strsql.SetString(mBuf.HinsyuCD,1);
	vaValuelist.PutElement(lArrayIndex, &(_variant_t(T2OLE(strsql))));

	lArrayIndex[0]=7;
	vaFieldlist.PutElement(lArrayIndex, &(_variant_t("KeiroCD")));	
	strsql.SetString(mBuf.KeiroCD,2);
	vaValuelist.PutElement(lArrayIndex, &(_variant_t(T2OLE(strsql))));

	lArrayIndex[0]=8;
	vaFieldlist.PutElement(lArrayIndex, &(_variant_t("SankuMochiKubun")));	
	strsql.SetString(mBuf.SankuMochiKubun,1);
	vaValuelist.PutElement(lArrayIndex, &(_variant_t(T2OLE(strsql))));

	lArrayIndex[0]=9;
	vaFieldlist.PutElement(lArrayIndex, &(_variant_t("ImportYear")));	
	strsql.SetString(mBuf.ImportYear,4);
	vaValuelist.PutElement(lArrayIndex, &(_variant_t(T2OLE(strsql))));

	lArrayIndex[0]=10;
	vaFieldlist.PutElement(lArrayIndex, &(_variant_t("BreederCode")));	
	strsql.SetString(mBuf.BreederCode,8);
	vaValuelist.PutElement(lArrayIndex, &(_variant_t(T2OLE(strsql))));

	lArrayIndex[0]=11;
	vaFieldlist.PutElement(lArrayIndex, &(_variant_t("SanchiName")));	
	strsql.SetString(mBuf.SanchiName,20);
	vaValuelist.PutElement(lArrayIndex, &(_variant_t(T2OLE(strsql))));

	lArrayIndex[0]=12;
	vaFieldlist.PutElement(lArrayIndex, &(_variant_t("FNum")));	
	strsql.SetString(mBuf.HansyokuNum[0],10);
	vaValuelist.PutElement(lArrayIndex, &(_variant_t(T2OLE(strsql))));

	lArrayIndex[0]=13;
	vaFieldlist.PutElement(lArrayIndex, &(_variant_t("MNum")));
	strsql.SetString(mBuf.HansyokuNum[1],10);
	vaValuelist.PutElement(lArrayIndex, &(_variant_t(T2OLE(strsql))));

	lArrayIndex[0]=14;
	vaFieldlist.PutElement(lArrayIndex, &(_variant_t("FFNum")));	
	strsql.SetString(mBuf.HansyokuNum[2],10);
	vaValuelist.PutElement(lArrayIndex, &(_variant_t(T2OLE(strsql))));

	lArrayIndex[0]=15;
	vaFieldlist.PutElement(lArrayIndex, &(_variant_t("FMNum")));	
	strsql.SetString(mBuf.HansyokuNum[3],10);
	vaValuelist.PutElement(lArrayIndex, &(_variant_t(T2OLE(strsql))));

	lArrayIndex[0]=16;
	vaFieldlist.PutElement(lArrayIndex, &(_variant_t("MFNum")));
	strsql.SetString(mBuf.HansyokuNum[4],10);
	vaValuelist.PutElement(lArrayIndex, &(_variant_t(T2OLE(strsql))));

	lArrayIndex[0]=17;
	vaFieldlist.PutElement(lArrayIndex, &(_variant_t("MMNum")));	
	strsql.SetString(mBuf.HansyokuNum[5],10);
	vaValuelist.PutElement(lArrayIndex, &(_variant_t(T2OLE(strsql))));

	lArrayIndex[0]=18;
	vaFieldlist.PutElement(lArrayIndex, &(_variant_t("FFFNum")));
	strsql.SetString(mBuf.HansyokuNum[6],10);
	vaValuelist.PutElement(lArrayIndex, &(_variant_t(T2OLE(strsql))));

	lArrayIndex[0]=19;
	vaFieldlist.PutElement(lArrayIndex, &(_variant_t("FFFNum")));	
	strsql.SetString(mBuf.HansyokuNum[7],10);
	vaValuelist.PutElement(lArrayIndex, &(_variant_t(T2OLE(strsql))));

	lArrayIndex[0]=20;
	vaFieldlist.PutElement(lArrayIndex, &(_variant_t("FMFNum")));
	strsql.SetString(mBuf.HansyokuNum[8],10);
	vaValuelist.PutElement(lArrayIndex, &(_variant_t(T2OLE(strsql))));

	lArrayIndex[0]=21;
	vaFieldlist.PutElement(lArrayIndex, &(_variant_t("FMFNum")));
	strsql.SetString(mBuf.HansyokuNum[9],10);
	vaValuelist.PutElement(lArrayIndex, &(_variant_t(T2OLE(strsql))));

	lArrayIndex[0]=22;
	vaFieldlist.PutElement(lArrayIndex, &(_variant_t("MFFNum")));	
	strsql.SetString(mBuf.HansyokuNum[10],10);
	vaValuelist.PutElement(lArrayIndex, &(_variant_t(T2OLE(strsql))));

	lArrayIndex[0]=23;
	vaFieldlist.PutElement(lArrayIndex, &(_variant_t("MFMNum")));
	strsql.SetString(mBuf.HansyokuNum[11],10);
	vaValuelist.PutElement(lArrayIndex, &(_variant_t(T2OLE(strsql))));

	lArrayIndex[0]=24;
	vaFieldlist.PutElement(lArrayIndex, &(_variant_t("MMFNum")));
	strsql.SetString(mBuf.HansyokuNum[12],10);
	vaValuelist.PutElement(lArrayIndex, &(_variant_t(T2OLE(strsql))));

	lArrayIndex[0]=25;
	vaFieldlist.PutElement(lArrayIndex, &(_variant_t("MMMNum")));	
	strsql.SetString(mBuf.HansyokuNum[13],10);
	vaValuelist.PutElement(lArrayIndex, &(_variant_t(T2OLE(strsql))));
	pRecordSet->Update(vaFieldlist, vaValuelist);
	pRecordSet->Close();	

}
catch(_com_error &e){
	MessageBox(NULL,e.Description(),NULL,NULL);
	pRecordSet->Close();
	return -1;
}

	return 0;

}
int clsImportSK::Init(_ConnectionPtr m_Connect)
{
    pCn = m_Connect;

	 return 0;
}

